compiler/rustc_lint/src/traits.rs

```diff
@@ -132,3 +134,43 @@ impl<'tcx> LateLintPass<'tcx> for DropTraitConstraints {
         }
     }
 }
+
+declare_lint! {
+    pub DUP_TRAIT_BOUNDS,
+    Deny,
+    "duplicate trait bounds"
+}
+
+declare_lint_pass!(
+    DuplicateTraitBounds => [DUP_TRAIT_BOUNDS]
+);
+
+impl<'tcx> LateLintPass<'tcx> for DuplicateTraitBounds {
+    fn check_ty(&mut self, cx: &LateContext<'_>, ty: &'tcx hir::Ty<'tcx>) {
```

Comment:
Why in `check_ty`, you are trying to check traits and trait aliases?  Those are items, so `check_item` is a better choice.

Reply:
Ah, thanks! I was just doing copy-paste from the code above. 😊